presto-cassandra/src/main/java/com/facebook/presto/cassandra/NativeCassandraSession.java

```diff
@@ -421,6 +432,30 @@ public ResultSet execute(Statement statement)
         return execute(partitionKeys).all();
     }
 
+    private Iterable<Row> queryPartitionKeysLegacy(CassandraTable table, List<Set<Object>> filterPrefixes)
```

Comment:
`SELECT * FROM x WHERE col1 IN (1,2,3) AND col2 IN ('a', 'b')` is equivalent to `SELECT * FROM x WHERE (col1 = 1 OR col1 = 2 OR col1 = 3) AND (col2 = 'a' OR col2 = 'b')`, maybe this would be better than executing 6 queries in Cassandra.

Reply:
@kokosing not sure if that's valid cql syntax.

Reply:
o_O I did not know that OR is not supported